### DISCUSSION TITLE
how do I install the thing? do I need to compile all of the rust files?

### DISCUSSION THREAD

**Discussion by hananelroe** [dmadisetti/steam-tui#44] [General]

**Comment by dmadisetti** [comment]
You have several options:

- The easiest is to use the binary provided under releases: https://github.com/dmadisetti/steam-tui/releases/tag/0.2.1
- If you are on Arch you can use the AUR package: https://aur.archlinux.org/packages/steam-tui
- If you are on Nix you can use the flake directly, or nixpkgs (though it may be old)
- If you are on something else and don't want to use the binary, you can compile and use this, by first [downloading cargo](https://doc.rust-lang.org/cargo/getting-started/installation.html):
  - and using the distribution on crates: https://crates.io/crates/steam-tui
  - OR running `cargo run` on the project. Note this option takes the most disk space, but is nice if you'd like to alter the program yourself.

**Reply by hananelroe** [reply]
when running the binary, using `./steam-tui` I get an error: `bash: ./steam-tui: No such file or directory`. do you have an idea why?

**Reply by dmadisetti** [reply]
Probably a missing library. You can do `ldd steam-tui` and it will tell you what libraries are found. Look for a line that says missing or not found.

**Reply by hananelroe** [reply]
libssl.so.1.1 and libcrypto.so.1.1 are indeed not found. do you know how to install them on Ubuntu 22.04? a google search didn't help

**Reply by dmadisetti** [reply]
`apt install libssl-dev` should address both:

 - https://packages.ubuntu.com/search?mode=exactfilename&suite=trusty&section=all&arch=any&keywords=libcrypto.so&searchon=contents
 - https://packages.ubuntu.com/search?mode=exactfilename&suite=trusty&section=all&arch=any&keywords=libssl.so&searchon=contents

**Reply by hananelroe** [reply]
Tried it, still not found

**Reply by dmadisetti** [reply]
Idk, are you sure there aren't any other missing libraries? You will also need `steamcmd`

The continuous integration runs `steam-tui` tests on ubuntu, and it still works. Maybe take a look? 

https://github.com/dmadisetti/steam-tui/blob/b56fef8498a7cf372c83c12ef9bd6cdb482f398f/.github/workflows/test.yml#L21

**Reply by hananelroe** [reply]
this is the output when trying to install lib32gcc1:

`Package lib32gcc1 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  lib32gcc-s1`

I installed lib32gcc-s1, but it didn't work either
